webui/react/src/components/kit/UserBadge.tsx

```diff
@@ -1,32 +1,24 @@
 import React from 'react';
 
-import Avatar, { Props as AvatarProps } from 'shared/components/Avatar';
-import useUI from 'shared/contexts/stores/UI';
-import { getDisplayName, UserNameFields } from 'utils/user';
+import { User } from 'types';
 
-import css from './UserBadge.module.scss';
+import Nameplate from './Nameplate';
+import UserAvatar from './UserAvatar';
 
-export interface Props extends Omit<AvatarProps, 'darkLight' | 'displayName'> {
+export interface Props {
   compact?: boolean;
-  user: UserNameFields;
+  hideAvatarTooltip?: boolean;
+  user?: User;
 }
 
-const UserBadge: React.FC<Props> = ({ user, compact, ...rest }) => {
-  const { ui } = useUI();
-  const displayName = getDisplayName(user);
-
-  const avatar = <Avatar {...rest} darkLight={ui.darkLight} displayName={displayName} />;
-  const classnames = [css.avatarCard];
-  if (compact) classnames.push(css.compact);
-
+const UserBadge: React.FC<Props> = ({ user, compact, hideAvatarTooltip }) => {
```

Comment:
Does this really need to be a component anymore? If so let's move it out of `kit`.

Reply:
Moving it out of `/kit` for now. 

Actually originally had this with `<UserBadge>` removed, but I ended up making it a wrapper around `<Nameplate>` since it does reduce duplication and it's consistent with `<UserAvatar>`, which is also just a user-specific wrapper around `<Avatar>`. Definitely open to removing it and going with passing the icon and text props directly, though.